pkg/controller/cloudmetrics/cloudmetrics_controller.go

```diff
@@ -77,47 +78,95 @@ var _ reconcile.Reconciler = &ReconcileCloudMetrics{}
+	postgresProviderList []providers.PostgresMetricsProvider
+	redisProviderList    []providers.RedisMetricsProvider
 }
 
 // Reconcile reads all redis and postgres crs periodically and reconcile metrics for these
 // resources.
 // The Controller will requeue the Request every 5 minutes constantly when RequeueAfter is set
 func (r *ReconcileCloudMetrics) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	r.logger.Info("Reconciling CloudMetrics")
+	r.logger.Info("reconciling CloudMetrics")
+	ctx := context.TODO()
 
 	// Fetch all redis crs
 	redisInstances := &integreatlyv1alpha1.RedisList{}
 	err := r.client.List(context.TODO(), redisInstances)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	if len(redisInstances.Items) > 0 {
-		for _, redis := range redisInstances.Items {
-			r.logger.Infof("Found redis cr: %s", redis.Name)
+	for _, redis := range redisInstances.Items {
+		r.logger.Infof("beginning to reconcile cloud metrics for redis cr: %s", redis.Name)
+		if err := r.reconcileRedisMetrics(ctx, &redis); err != nil {
+			r.logger.Error(err)
 		}
-	} else {
-		r.logger.Info("Found no redis instances")
 	}
 
 	// Fetch all postgres crs
 	postgresInstances := &integreatlyv1alpha1.PostgresList{}
 	err = r.client.List(context.TODO(), postgresInstances)
 	if err != nil {
-		return reconcile.Result{}, err
+		r.logger.Error(err)
 	}
-	if len(postgresInstances.Items) > 0 {
-		for _, postgres := range postgresInstances.Items {
-			r.logger.Infof("Found postgres cr: %s", postgres.Name)
+	for _, postgres := range postgresInstances.Items {
+		r.logger.Infof("beginning to reconcile cloud metrics for postgres cr: %s", postgres.Name)
+		if err := r.reconcilePostgresMetrics(ctx, &postgres); err != nil {
+			r.logger.Error(err)
 		}
-	} else {
-		r.logger.Info("Found no postgres instances")
 	}
 
-	// Requeue every 5 minutes
+	// we want full control over when we scrape metrics
+	// to allow for this we only have a single requeue
+	// this ensures regardless of errors or return times
+	// all metrics are scraped and exposed at the same time
 	return reconcile.Result{
-		RequeueAfter: watchDuration * time.Second,
+		RequeueAfter: resources.GetMetricReconcileTimeOrDefault(resources.MetricsWatchDuration),
 	}, nil
 }
+
+func (r *ReconcileCloudMetrics) reconcilePostgresMetrics(ctx context.Context, postgres *integreatlyv1alpha1.Postgres) error {
+	for _, p := range r.postgresProviderList {
+		// only scrape metrics on supported strategies
+		if !p.SupportsStrategy(postgres.Status.Strategy) {
+			continue
+		}
+
+		// all postgres metric providers inherit the same interface
+		// scrapeMetrics returns scraped metrics output which contains a list of GenericCloudMetrics
+		scrapedMetricsOutput, err := p.ScrapeRDSMetrics(ctx, postgres)
+		if err != nil {
+			r.logger.Errorf("failed to scrape metrics for postgres %v", err)
+			continue
+		}
+
+		for _, metricData := range scrapedMetricsOutput.Metrics {
+			r.logger.Debug(*metricData)
+		}
```

Comment:
I'm guessing we're planning to do something with these metrics (e.g. return them out of here) in a future ticket?

Reply:
at this point this is mainly here for verification, but I do see future uses for it.